File: AndroidStudioProjects/HealLotsMobile/app/src/main/java/com/heallots/mobile/ui/adapters/DashboardAppointmentAdapter.java
package com.heallots.mobile.ui.adapters;

import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.GradientDrawable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.heallots.mobile.R;
import com.heallots.mobile.models.Appointment;

import java.util.ArrayList;
import java.util.List;

public class DashboardAppointmentAdapter extends RecyclerView.Adapter<DashboardAppointmentAdapter.AppointmentViewHolder> {
    private final Context context;
    private final OnAppointmentClickListener listener;
    private List<Appointment> appointments = new ArrayList<>();

    public interface OnAppointmentClickListener {
        void onAppointmentClick(Appointment appointment);
    }

    public DashboardAppointmentAdapter(Context context, List<Appointment> appointments, OnAppointmentClickListener listener) {
        this.context = context;
        this.listener = listener;
        this.appointments = appointments != null ? appointments : new ArrayList<>();
    }

    public void updateAppointments(List<Appointment> updatedAppointments) {
        this.appointments = updatedAppointments != null ? updatedAppointments : new ArrayList<>();
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public AppointmentViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_dashboard_appointment, parent, false);
        return new AppointmentViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull AppointmentViewHolder holder, int position) {
        holder.bind(appointments.get(position));
    }

    @Override
    public int getItemCount() {
        return appointments.size();
    }

    class AppointmentViewHolder extends RecyclerView.ViewHolder {
        private final TextView emoji;
        private final TextView service;
        private final TextView specialist;
        private final TextView dateTime;
        private final TextView status;

        AppointmentViewHolder(@NonNull View itemView) {
            super(itemView);
            emoji = itemView.findViewById(R.id.dashboardAppointmentEmoji);
            service = itemView.findViewById(R.id.dashboardAppointmentService);
            specialist = itemView.findViewById(R.id.dashboardAppointmentSpecialist);
            dateTime = itemView.findViewById(R.id.dashboardAppointmentDateTime);
            status = itemView.findViewById(R.id.dashboardAppointmentStatus);
        }

        void bind(Appointment appointment) {
            service.setText(safeText(appointment.getServiceName(), "Hilot Session"));
            specialist.setText("with " + safeText(appointment.getSpecialistName(), "Specialist"));
            dateTime.setText("\uD83D\uDCC5 " + safeText(appointment.getAppointmentDate(), "TBD") + "   \u23F0 " + safeText(appointment.getTimeSlot(), "TBD"));
            emoji.setText(getEmoji(appointment.getServiceName()));
            status.setText(safeText(appointment.getStatus(), "Pending"));
            status.setBackground(makeStatusBackground(appointment.getStatus()));
            itemView.setOnClickListener(v -> {
                if (listener != null) {
                    listener.onAppointmentClick(appointment);
                }
            });
        }

        private GradientDrawable makeStatusBackground(String statusValue) {
            GradientDrawable drawable = new GradientDrawable();
            drawable.setCornerRadius(999f);
            String status = statusValue == null ? "" : statusValue.toLowerCase();
            if (status.contains("cancel")) {
                drawable.setColor(Color.parseColor("#FEE2E2"));
            } else if (status.contains("done") || status.contains("complete") || status.contains("approved")) {
                drawable.setColor(Color.parseColor("#DCFCE7"));
            } else if (status.contains("resched")) {
                drawable.setColor(Color.parseColor("#E0E7FF"));
            } else {
                drawable.setColor(Color.parseColor("#FEF3C7"));
            }
            return drawable;
        }

        private String getEmoji(String serviceName) {
            if (serviceName == null) {
                return "\uD83C\uDF3F";
            }
            switch (serviceName) {
                case "Traditional Hilot":
                    return "\uD83E\uDD32\uD83C\uDFFB";
                case "Herbal Compress":
                    return "\uD83C\uDF3F";
                case "Head & Neck Relief":
                    return "\uD83D\uDC86";
                case "Foot Reflexology":
                    return "\uD83E\uDDB6";
                case "Hot Oil Massage":
                    return "\uD83E\uDED9";
                case "Whole-Body Hilot":
                    return "\uD83E\uDDD8\uD83C\uDFFB";
                default:
                    return "\uD83C\uDF3F";
            }
        }

        private String safeText(String value, String fallback) {
            return value == null || value.trim().isEmpty() ? fallback : value;
        }
    }
}

File: AndroidStudioProjects/HealLotsMobile/app/src/main/java/com/heallots/mobile/ui/adapters/DashboardAppointmentAdapter.kt
package com.heallots.mobile.features.appointments.dashboard

import android.content.Context
import android.graphics.Color
import android.graphics.drawable.GradientDrawable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.heallots.mobile.R
import com.heallots.mobile.models.Appointment
import java.util.Locale

class DashboardAppointmentAdapter(
    private val context: Context,
    appointments: List<Appointment>?,
    private val listener: OnAppointmentClickListener?
) : RecyclerView.Adapter<DashboardAppointmentAdapter.AppointmentViewHolder>() {
    private var appointments: List<Appointment> = appointments ?: ArrayList()

    interface OnAppointmentClickListener {
        fun onAppointmentClick(appointment: Appointment)
    }

    fun updateAppointments(updatedAppointments: List<Appointment>?) {
        appointments = updatedAppointments ?: ArrayList()
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AppointmentViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_dashboard_appointment, parent, false)
        return AppointmentViewHolder(view)
    }

    override fun onBindViewHolder(holder: AppointmentViewHolder, position: Int) {
        holder.bind(appointments[position])
    }

    override fun getItemCount(): Int = appointments.size

    inner class AppointmentViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val emoji: TextView = itemView.findViewById(R.id.dashboardAppointmentEmoji)
        private val service: TextView = itemView.findViewById(R.id.dashboardAppointmentService)
        private val specialist: TextView = itemView.findViewById(R.id.dashboardAppointmentSpecialist)
        private val dateTime: TextView = itemView.findViewById(R.id.dashboardAppointmentDateTime)
        private val status: TextView = itemView.findViewById(R.id.dashboardAppointmentStatus)

        fun bind(appointment: Appointment) {
            service.text = safeText(appointment.serviceName, "Hilot Session")
            specialist.text = "with ${safeText(appointment.specialistName, "Specialist")}"
            dateTime.text = "\uD83D\uDCC5 ${safeText(appointment.appointmentDate, "TBD")}   \u23F0 ${safeText(appointment.timeSlot, "TBD")}"
            emoji.text = getEmoji(appointment.serviceName)
            status.text = safeText(appointment.status, "Pending")
            status.background = makeStatusBackground(appointment.status)
            itemView.setOnClickListener { listener?.onAppointmentClick(appointment) }
        }

        private fun makeStatusBackground(statusValue: String?): GradientDrawable {
            return GradientDrawable().apply {
                cornerRadius = 999f
                val status = statusValue?.lowercase(Locale.getDefault()).orEmpty()
                setColor(
                    when {
                        "cancel" in status -> Color.parseColor("#FEE2E2")
                        "done" in status || "complete" in status || "approved" in status -> Color.parseColor("#DCFCE7")
                        "resched" in status -> Color.parseColor("#E0E7FF")
                        else -> Color.parseColor("#FEF3C7")
                    }
                )
            }
        }

        private fun getEmoji(serviceName: String?): String {
            return when (serviceName) {
                "Traditional Hilot" -> "\uD83E\uDD32\uD83C\uDFFB"
                "Herbal Compress" -> "\uD83C\uDF3F"
                "Head & Neck Relief" -> "\uD83D\uDC86"
                "Foot Reflexology" -> "\uD83E\uDDB6"
                "Hot Oil Massage" -> "\uD83E\uDED9"
                "Whole-Body Hilot" -> "\uD83E\uDDD8\uD83C\uDFFB"
                else -> "\uD83C\uDF3F"
            }
        }

        private fun safeText(value: String?, fallback: String): String {
            return if (value.isNullOrBlank()) fallback else value
        }
    }
}
